libid/fractals.cpp

```diff
@@ -669,6 +733,95 @@ void mandel_perturb(
     delta_n.real(dnr);
 }
 
+double diff_abs(const double c, const double d)
+{
+    double cd = c + d;
+
+    if (c >= 0.0)
+    {
+        if (cd >= 0.0)
+            return d;
+        else
+            return -d - 2.0 * c;
+    }
+    else
+    {
+        if (cd > 0.0)
+            return d + 2.0 * c;
+        else
+            return -d;
+    }
+}
+
+void burning_ship_perturb(
+    const std::complex<double> &ref, std::complex<double> &delta_n, const std::complex<double> &delta0)
+{
+    int degree = (int) g_params[2];
+    const double r{ref.real()};
+    const double i{ref.imag()};
+    const double r2 = r * r;
+    const double i2 = i * i;
+    const double a{delta_n.real()};
+    const double b{delta_n.imag()};
+    const double a2 = a * a;
+    const double b2 = b * b;
+    const double a0{delta0.real()};
+    const double b0{delta0.imag()};
+    double dnr, dni, ab;
+
+    switch (degree)
+    {
+        case 2:
+            delta_n.real(2.0 * a * r + a2 - 2.0 * b * i - b2);
+            delta_n.imag(diff_abs(r * i, r * b + i * a + a * b) * 2);
+            delta_n += delta0;
+            break;
+        case 3: // Cubic Burning Ship
```

Comment:
drop the comments on the case statements as they just repeat what the code already says